tools/scripts/Makefile

```diff
@@ -0,0 +1,265 @@
+export PROCESSOR_CORE 
+export HARDWARE
+export PLATFORM
+
+NOOS_DIR 	  = $(CURDIR)/../..
+ROOT_DIR 	  = ../..
+DRIVERS_DIR 	  = $(NOOS_DIR)/drivers
+INCLUDE_DIR 	  = $(NOOS_DIR)/include
+SCRIPTS_DIR	  = $(NOOS_DIR)/tools/scripts
+PROJECTS_DIR 	  = $(NOOS_DIR)/projects
+SDK_WORKSPACE	  = $(PROJECT_DIR)/build
+BUILD_DIR	  = $(PROJECT_DIR)/build
+OBJECTS_DIR	  = $(BUILD_DIR)/obj
+SCRIPTS_PATH 	  = $(SCRIPTS_DIR)/platform/$(PLATFORM)
+LIBRARY_DIR 	  = $(BUILD_DIR)/bsp/$(PROCESSOR_CORE)/lib
+PLATFORM_DRIVERS  = $(DRIVERS_DIR)/platform/LOCAL_PLATFORM
+PROJECT_DIR 	  = $(PROJECTS_DIR)/$(PROJECT)
+
+# uncomment this line to include a custom config
+#include $(SCRIPTS_DIR)/config.mk
+include $(PROJECTS_DIR)/$(PROJECT)/src.mk
+
+# Default verbosity level is 0
+VERBOSE ?=0
+ifeq ($(VERBOSE),1)
+MUTE =
+NULL = 
+else
+MUTE = @
+NULL = >/dev/null
+endif
+
+EXEC = release
+TIMESTAMP = $(shell date +"%T")
+ECHO = @echo -n "\\e[32m[$(TIMESTAMP)]\\e[39m"
+ERROR = @echo -n "\\e[32m[$(TIMESTAMP)]\\e[31m"
+CC ?= gcc
+
+# Xilinx platform handling
+ifeq (xilinx,$(strip $(PLATFORM)))
+
+CFLAGS = -Wall 								\
+	 -Wmissing-field-initializers					\
+	 -Wclobbered 							\
+	 -Wempty-body 							\
+	 -Wignored-qualifiers 						\
+	 -Wmissing-parameter-type					\
+	 -Wold-style-declaration					\
+	 -Woverride-init 						\
+	 -Wsign-compare							\
+	 -Wtype-limits							\
+	 -Wuninitialized						\
+	 -Wunused-but-set-parameter					\
+	 -Wno-unused-parameter						\
+	 -fdata-sections						\
+	 -ffunction-sections 						\
+	 -std=gnu90							\
+	 -O2 								\
+	 -g3								\
+	 -MMD 								\
+	 -MP								\
+	 -lm						
+	#-Werror
+
+LDFLAGS = -Wl,-T							\
+	  -Wl,$(BUILD_DIR)/app/src/lscript.ld -L $(LIBRARY_DIR)
+
+# Zynq
+ifeq (ps7_cortexa9_0,$(strip $(PROCESSOR_CORE)))
+
+CC = arm-none-eabi-gcc
```

Comment:
it's a good idea to do `CC := arm-none-eabi-gcc` here; to make sure that it gets overriden;
there could be a chance that `CC` has `gcc` assigned;